File: 350-intersection-of-two-arrays-ii/intersection-of-two-arrays-ii.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        lookup = Counter(nums1)
        output = []

        for num in nums2:
            if num in lookup:
                if (lookup[num] > 0):
                    output.append(num)
                    lookup[num] -= 1

        return output
        

File: 350-intersection-of-two-arrays-ii/intersection-of-two-arrays-ii.js
var intersect = function(nums1, nums2) {
    let lookup = {}
    let output = []

    for (let i = 0; i < nums1.length; i++) {
        let char = nums1[i]
        if (!(char in lookup)) lookup[char] = 0
        lookup[char] ++ 
    }

    for (let i = 0; i < nums2.length; i++) {
        let char = nums2[i]
        if (!lookup[char]) continue
        output.push(char)
        lookup[char] --
    }    
    return output
};